### PR TITLE
[DO NOT MERGE] Update to msal 3.0.4 and some msbuild refactorings

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: active-directory-wpf-msgraph-v2/active-directory-wpf-msgraph-v2.csproj [hunk 1]
```diff
@@ -34,9 +34,6 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Microsoft.Identity.Client, Version=3.0.3-preview.0, Culture=neutral, PublicKeyToken=0a613f4dd989e8ae, processorArchitecture=MSIL">
```
Comment (bgavrilMS) [inline active-directory-wpf-msgraph-v2/active-directory-wpf-msgraph-v2.csproj]: Migrated to package reference using VS tool.

File: active-directory-wpf-msgraph-v2/MainWindow.xaml.cs [hunk 2]
```diff
@@ -49,8 +50,9 @@ private async void CallGraphButton_Click(object sender, RoutedEventArgs e)
 
                 try
                 {
-                    authResult = await app.AcquireTokenInteractive(scopes, this)
+                    authResult = await app.AcquireTokenInteractive(scopes)
                         .WithAccount(accounts.FirstOrDefault())
+                        .WithParentActivityOrWindow(new WindowInteropHelper(this).Handle) // optional, used to center the browser on the window
```
Comment (jmprieur) [inline active-directory-wpf-msgraph-v2/MainWindow.xaml.cs]: Can't we just pass this?

Comment (bgavrilMS) [reply active-directory-wpf-msgraph-v2/MainWindow.xaml.cs]: Yes, MSAL already takes a dependency on WinForms (and btw, website scenario customers are suprised that it does). We don't want to take a depnency on System.Windows (WPF) too.

Comment (bgavrilMS) [reply active-directory-wpf-msgraph-v2/MainWindow.xaml.cs]: Btw I tested with the handle and the centering algorithm works well.

Comment (bgavrilMS) [reply active-directory-wpf-msgraph-v2/MainWindow.xaml.cs]: If we want to the sample to be simpler, we can just remove it.

Comment (jmprieur) [reply active-directory-wpf-msgraph-v2/MainWindow.xaml.cs]: No, that's great.
We just need to update the wiki

**Pull request merged**